Mathlib/Analysis/Complex/Conformal.lean

```diff
@@ -159,3 +159,149 @@ theorem conformalAt_iff_differentiableAt_or_differentiableAt_comp_conj {f : ℂ
+derivative equals `ContinuousLinearMap.toComplexOfMapI` of the real derivative.
+-/
+theorem complexOfReal_fderivWithin (h₁ : DifferentiableWithinAt ℝ f s x)
+    (h₂ : fderivWithin ℝ f s x I = I • fderivWithin ℝ f s x 1) (hs : UniqueDiffWithinAt ℝ s x) :
+    fderivWithin ℂ f s x = (fderivWithin ℝ f s x).complexOfReal h₂ := by
+  have := ((differentiableWithinAt_complex_iff_differentiableWithinAt_real hs).2
+      ⟨h₁, h₂⟩).restrictScalars_fderivWithin ℝ hs
+  simpa [DFunLike.ext_iff]
+
+/--
+In cases where the Cauchy-Riemann Equation guarantees complex differentiability, the complex
+derivative equals `ContinuousLinearMap.toComplexOfMapI` of the real derivative.
+-/
+theorem complexOfReal_hasDerivWithinAt (h₁ : DifferentiableWithinAt ℝ f s x)
+    (h₂ : fderivWithin ℝ f s x I = I • fderivWithin ℝ f s x 1) :
+    HasDerivWithinAt f ((fderivWithin ℝ f s x).complexOfReal h₂ 1) s x := by
+  rw [hasDerivWithinAt_iff_hasFDerivWithinAt, smulRight_one_one]
+  exact HasFDerivWithinAt.complexOfReal_hasFDerivWithinAt h₁.hasFDerivWithinAt h₂
+
+/--
+The Cauchy-Riemann Equation: A real-differentiable function `f` on `ℂ` is complex-differentiable if
+and only if the derivative `fderiv ℝ f x` maps `I` to `I • (fderiv ℝ f x) 1`.
+-/
+theorem differentiableAt_complex_iff_differentiableAt_real :
+    DifferentiableAt ℂ f x ↔ DifferentiableAt ℝ f x ∧
+      fderiv ℝ f x I = I • fderiv ℝ f x 1 :=
+  ⟨fun h ↦ by simp [h.restrictScalars ℝ, h.fderiv_restrictScalars ℝ],
+    fun ⟨h₁, h₂⟩ => (differentiableAt_iff_restrictScalars ℝ h₁).2
+    ⟨(fderiv ℝ f x).complexOfReal h₂, rfl⟩⟩
+
+/--
+In cases where the Cauchy-Riemann Equation guarantees complex differentiability, the complex
+derivative equals `ContinuousLinearMap.toComplexOfMapI` of the real derivative.
+-/
+protected theorem HasFDerivAt.complexOfReal_hasFDerivAt {f' : ℂ →L[ℝ] E}
+    (h₁ : HasFDerivAt f f' x) (h₂ : f' I = I • f' 1) :
+    HasFDerivAt f (f'.complexOfReal h₂) x :=
+  hasFDerivAt_of_restrictScalars ℝ h₁ rfl
+
+/--
+In cases where the Cauchy-Riemann Equation guarantees complex differentiability, the complex
+derivative equals `ContinuousLinearMap.toComplexOfMapI` of the real derivative.
+-/
+theorem complexOfReal_hasDerivAt (h₁ : DifferentiableAt ℝ f x)
+    (h₂ : fderiv ℝ f x I = I • fderiv ℝ f x 1) :
+    HasDerivAt f ((fderiv ℝ f x).complexOfReal h₂ 1) x := by
+  rw [hasDerivAt_iff_hasFDerivAt, smulRight_one_one]
+  exact hasFDerivAt_of_restrictScalars ℝ h₁.hasFDerivAt rfl
+
+/--
+In cases where the Cauchy-Riemann Equation guarantees complex differentiability, the complex
+derivative equals `ContinuousLinearMap.toComplexOfMapI` of the real derivative.
+-/
+theorem complexOfReal_fderiv (h₁ : DifferentiableAt ℝ f x)
+    (h₂ : fderiv ℝ f x I = I • fderiv ℝ f x 1) :
+    (fderiv ℝ f x).complexOfReal h₂ = fderiv ℂ f x := by
+  have := (differentiableAt_complex_iff_differentiableAt_real.2 ⟨h₁, h₂⟩).fderiv_restrictScalars ℝ
```

Comment:
Can you simplify this using [HasFDerivAt.fderiv](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/Calculus/FDeriv/Basic.html#HasFDerivAt.fderiv)? Hopefully, this should be immediate from the lemma you proved above.